tests/test_shuffle.py

```diff
@@ -1,12 +1,59 @@
+import warnings
+
 import dask
+import pytest
+
+BASE_URL = "s3://dask-io/stability"
+storage_options = {"config_kwargs": {"region_name": "us-east-1"}}
+
+
+@pytest.fixture
+def shuffle_dataset(small_client):
+    # NOTE(sjperkins)
+    # I think the dataset and worker memory limits are related for this test
+    # as the dataset is large enough to provoke some spilling when it is
+    # persisted. Warn if the worker memory limits change egregiously.
+    # Alternatively, a separate cluster fixture may be appropriate
+    # for shuffle tests.
+    workers = [w for w in small_client.scheduler_info()["workers"].values()]
+    mem_limit = [w["memory_limit"] for w in workers.values()]
+    ONE_GB = 1024.0**3
+
+    if any(wm < 7.5 * ONE_GB for wm in mem_limit):
+        warnings.warn("Worker memory limit < 7.5GB")
 
+    if any(wm > 8.5 * ONE_GB for wm in mem_limit):
+        warnings.warn("Worker memory limit > 8.5GB")
+
+    if sum(mem_limit) > 85 * ONE_GB:
+        warnings.warn("Total worker memory limit > 85GB")
+
+    if sum(mem_limit) < 75 * ONE_GB:
+        warnings.warn("Total worker memory limit < 75GB")
 
-def test_shuffle_simple(small_client):
     df = dask.datasets.timeseries(
-        start="2000-01-01", end="2000-12-31", freq="1s", partition_freq="1D"
+        start="2000-01-01", end="2000-12-31", freq="50ms", partition_freq="1D"
     )
 
-    sdf = df.shuffle(on="x")
+    yield df
+    # url = f"{BASE_URL}/test-data"
+    # write = df.to_parquet(
+    #     url, compute=False, overwrite=True, storage_options=storage_options
+    # )
+    # size = df.memory_usage(index=True).sum() / ONE_GB
+
+    # size, _ = dask.compute(write, size)
+
+    # yield dd.read_parquet(url, storage_options=storage_options)
+
 
-    sdf_size = sdf.memory_usage(index=True).sum() / (1024.0**3)
-    assert sdf_size.compute() > 1.0  # 1.0 GB
+def test_shuffle_simple(shuffle_dataset):
+    sdf = shuffle_dataset.shuffle(on="x")
+    dask.compute(sdf)
```

Comment:
Beware, dask.compute brings everything into memory.  I would expect this to fail.

Reply:
It was succeeding in the jupyter notebook I was testing on, although there was a lot of spillage. In fact, this must be on an AWS bucket somewhere?

BTW I'm emphasising the AWS bucket because I could never get

```python
import tempfile

with tempfile.TemporaryDirectory() as tmpdirname:
  df.to_parquet(tmpdirname, ...)
```

to work (it complains it can't open one of the partitioned parquet files) and I've assumed that's because one doesn't necessarily have access to local store. However, if the worker's are spilling to disk then they must have access to some AWS file system. Any tips to unblock me here?

Reply:
So, `dask.compute(df.to_parquet())` will be just fine because the output is small (a bunch of filenames)

`dask.compute(df)` will fail because the output is big (a lot of actual data)

Reply:
I recommend proceding with this PR as if you had write access.  CI may still work (CI may have write access).  If it doesn't, then we'll ask other folks (like @ncclementi ) to make it so.  Let's give her something to work with though that is actually what we want to work.

Reply:
PR updated now that it has AWS read/write access